src/gei/Commands/MigrateArchiveRepoCommand.cs

```diff
@@ -0,0 +1,186 @@
+                azureStorageConnectionString = _environmentVariableProvider.AzureStorageConnectionString();
+
+                if (string.IsNullOrWhiteSpace(azureStorageConnectionString))
+                {
+                    throw new OctoshiftCliException("Please set either --azure-storage-connection-string or AZURE_STORAGE_CONNECTION_STRING");
+                }
+            }
+
+            if (noSslVerify)
+            {
+                _log.LogInformation("SSL verification disabled");
+            }
+
+            var githubApi = noSslVerify ? _sourceGithubApiFactory.CreateClientNoSsl(ghesApiUrl) : _sourceGithubApiFactory.Create(ghesApiUrl);
+            var azureApi = noSslVerify ? _azureApiFactory.CreateClientNoSsl(azureStorageConnectionString) : _azureApiFactory.Create(azureStorageConnectionString);
+
+            var gitDataArchiveId = await githubApi.StartGitArchiveGeneration(githubSourceOrg, sourceRepo);
+            _log.LogInformation($"Archive generation of git data started with id: {gitDataArchiveId}");
+            var metadataArchiveId = await githubApi.StartMetadataArchiveGeneration(githubSourceOrg, sourceRepo);
+            _log.LogInformation($"Archive generation of metadata started with id: {metadataArchiveId}");
+
+            var metadataArchiveUrl = await WaitForArchiveGeneration(githubApi, githubSourceOrg, metadataArchiveId);
+            _log.LogInformation($"Archive (metadata) download url: {metadataArchiveUrl}");
+            var gitArchiveUrl = await WaitForArchiveGeneration(githubApi, githubSourceOrg, gitDataArchiveId);
+            _log.LogInformation($"Archive (git) download url: {gitArchiveUrl}");
+
+            var timeNow = $"{dateTimeNow:yyyy-MM-dd_HH-mm-ss}";
+
+            var gitArchiveFileName = $"{timeNow}-{GIT_ARCHIVE_FILE_NAME}";
+            var metadataArchiveFileName = $"{timeNow}-{METADATA_ARCHIVE_FILE_NAME}";
+
+            // Download both archives to the local filesystem
+            _log.LogInformation($"Downloading archive from {gitArchiveUrl}");
+            var gitArchiveContent = await azureApi.DownloadArchive(gitArchiveUrl);
+            _log.LogInformation($"Downloading archive from {metadataArchiveUrl}");
+            var metadataArchiveContent = await azureApi.DownloadArchive(metadataArchiveUrl);
+
+            _log.LogInformation($"Uploading archive {gitArchiveFileName} to Azure Blob Storage");
+            var authenticatedGitArchiveUrl = await azureApi.UploadToBlob(gitArchiveFileName, gitArchiveContent);
+            _log.LogInformation($"Uploading archive {metadataArchiveFileName} to Azure Blob Storage");
+            var authenticatedMetadataArchiveUrl = await azureApi.UploadToBlob(metadataArchiveFileName, metadataArchiveContent);
+
+            // Run migrate repo command
+            var migrateRepoCommand = new MigrateRepoCommand(_log, _targetGithubApiFactory, _environmentVariableProvider);
```

Comment:
We'll merge this command into the migrate repo command in a followup and then add unit tests. We ran into errors trying to add the migrate repo command through dependency injection